OpenEdXMobile/src/main/java/org/edx/mobile/util/FileUtil.java

```diff
@@ -83,17 +83,9 @@ public static void deleteRecursive(@NonNull File fileOrDirectory,
      * @param context The current context.
      * @return App's external storage directory.
      */
-    @NonNull
+    @Nullable
     public static File getAppExternalDir(@NonNull Context context) {
         File externalFilesDir = context.getExternalFilesDir(null);
-        if (externalFilesDir != null) {
-            return externalFilesDir.getParentFile();
-        } else {
-            File android = new File(Environment.getExternalStorageDirectory(), "Android");
-            File downloadsDir = new File(android, "data");
-            File packageDir = new File(downloadsDir, context.getPackageName());
-            packageDir.mkdirs();
-            return packageDir;
-        }
+        return (externalFilesDir != null ? externalFilesDir.getParentFile() : null);
```

Comment:
2 Reasons they are there:

- To improve readability.
- I favor this way of encapsulating a ternary operation.